src/chess/bitboard.h

```diff
@@ -190,40 +190,64 @@ class Move {
  public:
   enum class Promotion : std::uint8_t { None, Queen, Rook, Bishop, Knight };
   Move() = default;
-  Move(BoardSquare from, BoardSquare to) : from_(from), to_(to) {}
-  Move(BoardSquare from, BoardSquare to, Promotion promotion)
-      : from_(from), to_(to), promotion_(promotion) {}
-  Move(const std::string& str, bool black = false);
-  Move(const char* str, bool black = false) : Move(std::string(str), black) {}
-
-  BoardSquare from() const { return from_; }
-  BoardSquare to() const { return to_; }
-  Promotion promotion() const { return promotion_; }
-  bool IsCastling() const { return castling_; }
-  void SetCastling() { castling_ = true; }
+  Move(BoardSquare from, BoardSquare to) {
+    SetTo(to);
+    SetFrom(from);
+  }
+  Move(BoardSquare from, BoardSquare to, Promotion promotion) {
+    SetTo(to);
+    SetFrom(from);
+    SetPromotion(promotion);
+  }
+  Move(const std::string &str, bool black = false);
+  Move(const char *str, bool black = false) : Move(std::string(str), black) {}
+
+  BoardSquare GetTo() const { return (BoardSquare)((data_ & TO_MASK_)); }
```

Comment:
Modern way is to write
`{data_ & TO_MASK_}` instead of `(BoardSquare)((data_ & TO_MASK_))`.

Reply:
That worked for the BoardSquares, but threw an error for the Promotion